apps/wallet-mobile/src/Staking/PoolTransition/usePoolTransition.tsx

```diff
@@ -0,0 +1,426 @@
+                poolId,
+                yoroiUnsignedTx,
+              },
+            },
+          },
+        })
+
+  return {
+    poolTransition,
+    canTransitionPool: !!yoroiUnsignedTx,
+    navigateToUpdate,
+    isLoading: poolTransitionQuery.isLoading || stakingTxQuery.isLoading,
+  }
+}
+
+export const usePoolTransitionModal = () => {
+  const {poolTransition, canTransitionPool, isLoading} = usePoolTransition()
+  const {openModal} = useModal()
+  const strings = useStrings()
+  const {modalHeight} = useStyles()
+
+  React.useEffect(() => {
+    if (canTransitionPool) {
+      openModal(strings.upgradeStakePool, <Modal />, modalHeight)
+    }
+  }, [canTransitionPool, modalHeight, openModal, poolTransition, strings.upgradeStakePool])
+
+  return {isLoading}
+}
+
+const Modal = () => {
+  const {poolTransition, navigateToUpdate} = usePoolTransition()
+
+  const {styles, colors} = useStyles()
+  const strings = useStrings()
+
+  const {closeModal} = useModal()
+
+  React.useEffect(() => {
+    if (!poolTransition) {
+      closeModal()
+    }
+  }, [closeModal, poolTransition, strings.upgradeStakePool])
+
+  if (!poolTransition) return null
+
+  const handleOnSkip = () => {
+    closeModal()
+  }
+
+  const handleOnUpdate = () => {
+    closeModal()
+    navigateToUpdate()
+  }
+
+  const isActive = poolTransition.deadlineMilliseconds > 0
+
+  return (
+    <View style={styles.modal}>
+      <Text style={styles.details}>{isActive ? strings.warning : strings.finalWarning}</Text>
+
+      <Spacer fill height={16} />
+
+      <View style={[styles.card, isActive ? styles.border : styles.redBorder]}>
+        <View style={styles.row}>
+          <Text style={[styles.regular, styles.gray]}>{strings.currentPool}</Text>
+
+          <View style={styles.tickerWrap}>
+            {poolTransition.currentPool.pic != null && (
+              <Image source={{uri: poolTransition.currentPool.pic}} style={styles.pic} />
+            )}
+
+            <Text
+              style={[styles.regular, styles.primary]}
+            >{`[${poolTransition.currentPool.ticker}] ${poolTransition.currentPool.name}`}</Text>
+          </View>
+        </View>
+
+        <View style={styles.row}>
+          <Text style={[styles.regular, styles.gray]}>{strings.estimatedRoa}</Text>
+
+          <Text style={[styles.regular, styles.max]}>{poolTransition.currentPool.roa} %</Text>
+        </View>
+
+        <View style={styles.row}>
+          <Text style={[styles.regular, styles.gray]}>{strings.fee}</Text>
+
+          <Text style={[styles.regular, styles.max]}>{poolTransition.currentPool.tax} %</Text>
+        </View>
+
+        <View>
+          <Text style={[styles.regular, styles.red]}>
+            {isActive ? strings.poolWillStopRewards : strings.poolNoRewards}
+          </Text>
+
+          {isActive && (
+            <Text style={[styles.red, styles.bold]}>{formatTimeSpan(poolTransition.deadlineMilliseconds)}</Text>
+          )}
+        </View>
```

Comment:
```suggestion
          <Text style={[styles.regular, styles.red]}>
            {isActive ? strings.poolWillStopRewards : strings.poolNoRewards}
          </Text>

          {isActive && (
            <Text style={[styles.red, styles.bold]}>{formatTimeSpan(poolTransition.deadlineMilliseconds)}</Text>
          )}
```

Reply:
Are you just removing the View? That would increase the gap between these 2 components.

Reply:
Because the parent component uses flex and gap to separate it's children. This is a single sentence divided in 2 Texts with different styles and one of them being optional. With a wrapper, both texts stay together. Without it, they'll be treated as separate children of the flex parent and be affected by the gap.

Reply:
it would be simpler to read if the Texts were contained by a Text, not by a View, to indicate a single line, if it was grabbing a ready text node from somewhere else it would make sense, but here it can be accomplished by just moving Text - inside Text.

Reply:
Sure, the wrapper can be another Text. I was thinking of html <p> which can't be nested.